### PR TITLE
[stable9.1] Add all properties while creating a subscription

### LINKED ISSUE #24469: CalDAVBackend::createSubscription ignores a lot of properties
https://github.com/owncloud/core/blob/master/apps/dav/lib/caldav/caldavbackend.php#L1063-1070

This code only inserts the hardcoded values. Properties like displayname, color, order, stripAlerts and stripAttachments are simply ignored without further notice.

createCalendar inserts all supported properties: https://github.com/owncloud/core/blob/master/apps/dav/lib/caldav/caldavbackend.php#L378-380

### PR DESCRIPTION
**Pull request by DeepDiver1975**
backport of  #25318

Fixes #24469

### CODE REVIEW

**Comment by PVince81** [comment]
not critical -> 9.1.1

**Comment by DeepDiver1975** [comment]
@georgehrke @PVince81 mind testing/review? THX

**Comment by georgehrke** [comment]
👍

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -1079,22 +1079,27 @@ function createSubscription($principalUri, $uri, array $properties) {
...
+		$propertiesBoolean = ['striptodos', 'stripalarms', 'stripattachments'];
 
-				$values[$dbName] = $properties[$xmlName];
-				$fieldNames[] = $dbName;
+		foreach($this->subscriptionPropertyMap as $xmlName=>$dbName) {
+			if (array_key_exists($xmlName, $properties)) {
+					$values[$dbName] = $properties[$xmlName];
+					if (in_array($dbName, $propertiesBoolean)) {
+						$values[$dbName] = true;
+				}
```
Comment (PVince81) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: the intend is off :see_no_evil:

Comment (PVince81) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: indent INDEEEENT

**Comment by PVince81** [comment]
Code looks good otherwise :+1:

**Comment by DeepDiver1975** [comment]
indent fixed -> merge

**Pull request merged**